api/reporting/schemas/reporting-codegen.rb

```diff
@@ -26,64 +27,187 @@ def to_output_path(dir_name, table_name, suffix)
     :swagger_yaml => to_output_path(File.join(output_dir, 'swagger_yaml'), table_name,'yaml'),
     :projection_interface => to_output_path(File.join(output_dir, 'projection_interface'), table_name, 'java'),
-    :projection_query => to_output_path(File.join(output_dir, 'projection_query'), table_name,'java')
-}
-
-MYSQL_TO_BIGQUERY_TYPE = {
-    'varchar' => 'STRING',
-    'datetime' => 'TIMESTAMP',
-    'bigint' => 'INT64',
-    'smallint' => 'INT64',
-    'longtext' => 'STRING',
-    'int' => 'INT64',
-    'tinyint' => 'INT64',
-    'bit' => 'BOOLEAN',
-    'double' => 'FLOAT64',
-    'text' => 'STRING',
-    'mediumblob' => 'STRING'
-}
-
-def to_bq_type(mysql_type)
+    :projection_query => to_output_path(File.join(output_dir, 'projection_query'), table_name,'java'),
+    :unit_test_constants => to_output_path(File.join(output_dir, 'unit_test_constants'), table_name, 'java'),
+    :unit_test_mocks => to_output_path(File.join(output_dir, 'unit_test_mocks'), table_name, 'java'),
+    :dto_assertions => to_output_path(File.join(output_dir, 'dto_assertions'), table_name, 'java'),
+    :query_parameter_COLUMNS => to_output_path(File.join(output_dir, 'query_parameter_COLUMNS'), table_name, 'java'),
+    :dto_decl => to_output_path(File.join(output_dir, 'dto_decl'), table_name, 'java'),
+    :entity_decl => to_output_path(File.join(output_dir, 'entity_decl'), table_name, 'java'),
+}.freeze
+
+# This is the canonical type map, but there are places where we assign a tinyint MySql column a Long Entity field, etc.
+MYSQL_TO_TYPES = {
+    'varchar' => {
+        :bigquery => 'STRING',
+        :java => 'String'
+    },
+    'datetime' => {
+        :bigquery => 'TIMESTAMP',
+        :java => 'Timestamp'
+    },
+    'bigint' => {
+        :bigquery => 'INT64',
+        :java => 'Long'
+    },
+    'smallint' => {
+        :bigquery => 'INT64',
+        :java => 'Short'
+    },
+    'longtext' => {
+        :bigquery => 'STRING',
+        :java => 'String'
+    },
+    'int' => {
+        :bigquery => 'INT64',
+        :java => 'Integer'
+    },
+    'tinyint' => {
+        :bigquery => 'INT64',
+        :java => 'Short'
+    },
+    'bit' => {
+        :bigquery => 'BOOLEAN',
+        :java => 'Boolean'
+    },
+    'double' => {
+        :bigquery =>  'FLOAT64',
+        :java => 'Double'
+    },
+    'text' => {
+        :bigquery => 'STRING',
+        :java => 'String'
+    },
+    'mediumblob' => {
+        :bigquery => 'STRING',
+        :java => 'String'
+    }
+}.freeze
+
+# Enumerated types should be the enum in Java and the DTO, but a STRING in BQ.
+# We could either try to remap to the enum type when writing to BQ, but we still
+# have to handle getting the value into the projection, and the only way I've been
+# able to do that is to use the exact same type.
+ENUM_TYPES = {
+     'workspace' => {
+        'billing_status' => {
+            :java => 'BillingStatus',
+            :constant => 'BillingStatus.ACTIVE'
```

Comment:
what is `:constant`? Is this a default value ?

Reply:
It's something for the right hand side in unit test constant declarations; nothing to do with the schema.

Reply:
I still don't understand, but please add a comment to the code